pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -34,139 +42,101 @@
 , xkeyboard_config
 , xorg
 , zlib
+# settings and optional packages
+, cudaSupport ? false
+, cudaPackages ? null
 , lang ? "en"
+, version ? null
 }:
 
-let
-  l10n = import ./l10ns.nix {
-    inherit lib requireFile lang;
-  };
-in stdenv.mkDerivation {
-  inherit (l10n) version name src;
+let versions = import ./versions.nix { inherit lib requireFile; };
+
+    matching-versions =
+      with lib; sort compareVersions (filter
+        (v: v.lang == lang
+            && (if version == null then true else isMatching v.version version))
+        versions);
+
+    found-version =
+      if matching-versions == []
+      then throw ("No registered Mathematica version found to match"
+                  + " version=${version} and language=${lang}")
```

Comment:
```suggestion
      then throw ("No registered Mathematica version found to match version=${version} and language=${lang}")
```

Reply:
Well, I am avoiding a single line exceeding 80 characters here.

Reply:
There is no 80 chars line limit in nixpkgs and this makes it easier to grep for the string.

Reply:
> Well, I am avoiding a single line exceeding 80 characters here.

This is a matter of taste IMO in which I think the package maintainer has the advantage.